storagenode/piecestore/endpoint.go

```diff
@@ -547,25 +548,37 @@ func (endpoint *Endpoint) SaveOrder(ctx context.Context, limit *pb.OrderLimit, o
 func (endpoint *Endpoint) Retain(ctx context.Context, retainReq *pb.RetainRequest) (*pb.RetainResponse, error) {
 	peer, err := identity.PeerIdentityFromContext(ctx)
 	if err != nil {
-		return nil, Error.Wrap(err)
+		return nil, status.Error(codes.Internal, Error.Wrap(err).Error())
```

Comment:
this is not internal

Reply:
what should it be instead then?

Reply:
https://github.com/grpc/grpc/blob/master/doc/statuscodes.md